### PR TITLE
application: refactor version checks, move to ApplicationUtils

### PR DESCRIPTION
**Pull request by virtualtam**
Relates to #372

Modifications:
 - move checkUpdate() to ApplicationUtils
 - reduce file I/O operations during version checks
 - apply coding conventions
 - add test coverage

Tools:
 - create a sandbox directory for tests

### CODE REVIEW

File: application/ApplicationUtils.php [hunk 1]
```diff
@@ -4,6 +4,97 @@
  */
 class ApplicationUtils
 {
+    private static $GIT_URL = 'https://raw.githubusercontent.com/shaarli/Shaarli';
+    private static $GIT_BRANCH = 'master';
```
Comment (ArthurHoaro) [inline application/ApplicationUtils.php]: Shouldn't we check the stable branch?

File: application/ApplicationUtils.php [hunk 2]
```diff
@@ -4,6 +4,97 @@
...
+     * - in case of failure, the update file's modification date is updated,
+     *   to avoid intempestive connection attempts.
+     *
+     * @param string $currentVersion the current version code
+     * @param string $updateFile     the file where to store the latest version code
+     * @param int    $checkInterval  the minimum interval between update checks (in seconds
+     * @param bool   $enableCheck    whether to check for new versions
+     * @param bool   $isLoggedIn     whether the user is logged in
+     *
+     * @return string the new version code if available and greater, '' else
```
Comment (ArthurHoaro) [inline application/ApplicationUtils.php]: Not very important, but I think that returning `false` makes more sense that returning a blank string.

**Comment by ArthurHoaro** [comment]
Great! I've made a few comments, not sure they're really relevant though.

File: application/ApplicationUtils.php (hunk 1 above)
Comment (virtualtam) [inline application/ApplicationUtils.php]: Yup, but I'd rather change this behavior in a (near-)future commit, so it shows in the Git log :)

File: application/ApplicationUtils.php (hunk 2 above)
Comment (virtualtam) [inline application/ApplicationUtils.php]: +1, logging can also be added when sending HTTP requests

**Comment by virtualtam** [comment]
Thanks! I'll update the PR accordingly :)

**Pull request merged**